cdi/src/main/java/io/smallrye/config/inject/InjectionMessages.java

```diff
@@ -10,36 +9,30 @@
 import org.jboss.logging.annotations.Message;
 import org.jboss.logging.annotations.MessageBundle;
 import org.jboss.logging.annotations.Param;
-import org.jboss.logging.annotations.Pos;
-import org.jboss.logging.annotations.Signature;
 
 @MessageBundle(projectCode = "SRCFG", length = 5)
 public interface InjectionMessages {
     InjectionMessages msg = Messages.getBundle(InjectionMessages.class);
 
-    @Signature(messageIndex = 1, value = { String.class, String.class })
-    @Message(id = 2000, value = "No Config Value exists for required property %s")
-    ConfigException noConfigValue(@Param @Pos(1) String key);
+    @Message(id = 2000, value = "Failed to Inject @ConfigProperty for key %s into %s since the config property could not be found in any config source")
+    ConfigInjectionException noConfigValue(String key, String location, @Param String configPropertyName);
 
-    @Signature(messageIndex = 1, value = { String.class, String.class, Throwable.class })
-    @Message(id = 2001, value = "Failed to retrieve config for key %s")
-    ConfigException retrieveConfigFailure(@Cause IllegalArgumentException cause, @Param @Pos(1) String key);
+    @Message(id = 2001, value = "Failed to Inject @ConfigProperty for key %s into %s %s")
+    ConfigInjectionException retrieveConfigFailure(String key, String location, String causeMessage,
+            @Param String configPropertyName, @Cause Exception e);
```

Comment:
If I'm reading this correctly, you now have to pass in the key twice (once as `key` and once as `configPropertyName`)? I guess this is because you've removed `@Param @Pos(1)` from `key`?

If you do need to pass in the same property twice, it would be good to note that as it's weird.

Reply:
+1